docs/source/csdl-spec.mdx

```diff
@@ -0,0 +1,319 @@
+
+The one difference to note is similar to the `@key` directive above: their `fields` arguments are bracketed.
+
+```graphql
+type Rocket 
+  @key(fields: "{ id }", graph: "rockets")
+  @key(fields: "{ id }", graph: "astronauts") 
+{
+	id: String! 
+  name: String!
+	astronaut: Astronaut!
+  captain: Astronaut @provides(fields: "{ tripId }")
+}
+```
+
+### `@owner`
+
+`directive @owner(graph: String!) on OBJECT`
+
+Just like any other GraphQL service, federated services can define any types they would like or need. When a service defines a type that hasn't been defined before (that is, it's not extending a type), then that service is referred to as the `owner` of that type. Type ownership is important, because services have the ability to extend types defined by other services in the graph, and those extenders can require additional fields from the owner to help them resolve their extensions. In our example, the `Astronaut` type is defined by the astronauts service, and the `Rocket`  type by the rockets service. We can represent that in CSDL like so:
+
+```graphql
+type Astronaut 
+	@key(fields: "{ id }", graph: "astronauts")
+	@key(fields: "{ id }", graph: "rockets")
+	@owner(graph: "astronauts")
+{
+  id: String!
+  name: String!
+  rocket: Rocket!
+	tripId: String!
+}
+
+type Rocket 
+	@key(fields: "{ id }", graph: "rockets")
+	@key(fields: "{ id }", graph: "astronauts")
+	@owner(graph: "rockets")
+{
+  id: String!
+  name: String!
+	astronaut: Astronaut!
+  captain: Astronaut @provides(fields: "{ tripId }")
+}
+```
+
+**Not all types** get an `@owner` defined in CSDL:
+- Value types are shared amongst all services and don't receive an owner.
+- Types which are defined and used only in one service don't receive an owner.
+- Root types (`Query` and `Mutation`) must not have owners.
+
+
+### `@resolve`
+
+`directive @resolve(graph: String!) on FIELD_DEFINITION`
+
+Once we know about owning services for types, we have to represent what service is responsible for resolving fields that are extended by other services. We do this with the `resolve` directive. For the `rocket` field on our `Astronaut` type and the `astronaut` field on our `Rocket` type, that would look like this:
+
+```graphql
+type Astronaut 
+	@key(fields: "{ id }", graph: "astronauts")
+	@key(fields: "{ id }", graph: "rockets")
+	@owner(graph: "astronauts")
```

Comment:
what do we think about changing `@owner(graph:)` to `@default(graph:)`? i'm thinking about the discussions we've been having about something in-between value types and owned types. In principle, the only thing that's special about a type's owner is that it's the default resolver for fields (and also that you can only `@key` off fields from the owner type, but I don't think that's a requirement within this spec, but rather within federation)

Reply:
@trevor-scheer @JakeDawkins @zionts ☝️